### PR TITLE
[SQL Migration] Add additional checks for multiple database backups in same folder 

### PR DESCRIPTION
**Pull request by raymondtruong**
A common top error is that backups for multiple databases are present in the same folder, which is not supported by MI / VM LRS. A change was added in https://github.com/microsoft/azuredatastudio/pull/21952 to allow the selection of individual folders in blob containers, but the occurrence of this error has not decreased significantly. This PR:
- adds an info box which instructs users to place backups from multiple databases into separate folders
   - public facing documentation still needs to be updated
- adds a check where, if migrating multiple databases at a time, each database backup location (combination of storage account, blob container, and folder name) should be unique, since this scenario shouldn't be possible (unless they're migrating duplicates of the same database?). The user sees a warning banner until each database has a unique backup location provided.

image

image

### CODE REVIEW

**Comment by austincraigbryan** [comment]
Shouldn't we block moving forward if we know it will fail under this condition

File: extensions/sql-migration/src/wizard/databaseBackupPage.ts [hunk 1]
```diff
@@ -1184,6 +1195,28 @@ export class DatabaseBackupPage extends MigrationWizardPage {
 										if (selectedLastBackupFile && !blobFileErrorStrings.includes(value)) {
 											this.migrationStateModel._databaseBackup.blobs[index].lastBackupFile = selectedLastBackupFile.name;
 										}
+
+										// check for duplicate storage account/blob container/last backup file combination if migrating multiple databases,
+										// as they should all be unique - backups for multiple databases in the same location are not supported
+										var backupLocations: string[] = [];
```
Comment (brian-harris) [inline extensions/sql-migration/src/wizard/databaseBackupPage.ts:1201]: should we prefer 
"let" over "var"

**Review by brian-harris — APPROVED** [review]

**Comment by raymondtruong** [comment]
> Shouldn't we block moving forward if we know it will fail under this condition. Instead of a warning it's almost more apt to be an error

@austincraigbryan 

hmm, yeah I guess so. My last PR (https://github.com/microsoft/azuredatastudio/pull/22410) I did as a warning because in that case you can't be 100% certain technically, but I guess in this case it should be safer. The only valid case I can think of is migrating the same database multiple times at once (in fact, like my screenshot), which is why I left it as non blocking, but that's about it. If that's it then is that good enough to call it a blocker?

**Comment by austincraigbryan** [comment]
> > Shouldn't we block moving forward if we know it will fail under this condition. Instead of a warning it's almost more apt to be an error
> 
> @austincraigbryan
> 
> hmm, yeah I guess so. My last PR (#22410) I did as a warning because in that case you can't be 100% certain technically, but I guess in this case it should be safer. The only valid case I can think of is migrating the same database multiple times at once (in fact, like my screenshot), which is why I left it as non blocking, but that's about it. If that's it then is that good enough to call it a blocker?

Hmm guess I would leave something like that up to a PM I'm not sure the validity of the scenario. Maybe David would have some input. I would be worried customers would just see they can go next once selecting their folders and move on without even paying attention

**Comment by brian-harris** [comment]
be sure not to block the ability to migrate the same database backups to multiple targets.

File: extensions/sql-migration/src/wizard/databaseBackupPage.ts (hunk 1 above)
Comment (raymondtruong) [reply extensions/sql-migration/src/wizard/databaseBackupPage.ts:1201]: thanks, I keep forgetting

**Pull request merged**